internal/ingestion/service/process_message_test.go

```
@@ -309,37 +310,146 @@ func TestProcessMessage_ClaimSucceedsEnqueues(t *testing.T) {
+	defer cleanup()
+
+	ingestor := NewIngestor("test", 1, []string{"pdf"})
+	ingestor.heartbeatInterval = 20 * time.Millisecond
+	ingestor.SetMemoryMessageService(service.NewMemoryMessageService(nil))
 
-	// Drain the fillers.
+	// maxConcurrency=1 → channel cap=2. Fill it completely so queueTaskCtx
+	// blocks for the memory task.
 	for i := 0; i < cap(ingestor.taskChan); i++ {
-		<-ingestor.taskChan
+		ingestor.taskChan <- taskpkg.NewTaskContextForScheduling(nil, &entity.IngestionTask{ID: "filler"})
```

golangci-lint (2.12.2)

[error] SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

(staticcheck)